repo/mk/js.deps.npm.mk

```diff
@@ -102,13 +102,19 @@ deps-npm-install:
 	[[ ! -f node_modules/eslint-config-firecloud/package.json ]] || \
 		$(SUPPORT_FIRECLOUD_DIR)/bin/npm-install-peer-deps \
 			node_modules/eslint-config-firecloud/package.json
-#	hack. sort dependencies in package.json
-	if $(CAT) package.json | $(GREP) -q "\"dependencies\""; then \
-		$(NPM) remove --save-prod some-pkg-that-doesnt-exist; \
-	fi
-	if $(CAT) package.json | $(GREP) -q "\"devDpendencies\""; then \
-		$(NPM) remove --save-dev some-pkg-that-doesnt-exist; \
-	fi
+#	sort dependencies in package.json
+	$(CAT) package.json | \
+		$(JQ) ". \
+			+ (if .dependencies \
+				then {dependencies: (.dependencies|to_entries|sort_by(.key)|from_entries)} \
+				else .dependencies \
+				end) \
+			+ (if .devDependencies \
+				then {devDependencies: (.devDependencies|to_entries|sort_by(.key)|from_entries)} \
+				else .devDependencies \
+				end) \
+		" | \
+		${SUPPORT_FIRECLOUD_DIR}/bin/sponge package.json
```

Comment:
I learned about sponge: "Unlike a shell redirect, sponge soaks up all its input before opening the output file." Thanks!
I don't understand the JQ script, but can can accept that it's doing it's thing: "jq is like sed for JSON data".
I still don't understand how this has a different effect than previous code, though. Both write a new package.json, where the dependencies have been ordered. What's the difference?

Reply:
"What's the difference?" Well, that's the bug...